### ISSUE TITLE
Build doc fails if nuget restore is not run

### ISSUE THREAD

**Issue by xen2** [dotnet/docfx#3177]
On our project ([Xenko](https://xenko.com)), if nuget restore is not ran prior to docfx, we have an NullReferenceException because `ErrorType` from `ErrorType ValueTuple<?>`  has no `ContainingNamespace` (null) so this check fail: https://github.com/dotnet/docfx/blob/v2.37.2/src/Microsoft.DocAsCode.Metadata.ManagedReference.Roslyn/Visitors/NameVisitor.cs#L128

So I have a few questions:
1. Are we supposed to call nuget restore ourselves before running docfx?
2. Is docfx supposed to work if nuget restore is not done by handling those `ErrorType`?

**DocFX Version Used**:  2.37.2

**Steps to Reproduce**:

1. Clone `https://github.com/xenko3d/xenko`  (do not open solution in VS)
2. Clone `https://github.com/xenko3d/xenko-docs` (so that `xenko` and `xenko-docs` are side by side)
3. Run `deps\docfx\docfx.exe metadata en/docfx.json` from `xenko-docs` folder

**Expected Behavior**:

Success

**Actual Behavior**:
```
Microsoft.DocAsCode.Exceptions.DocfxException: Unable to generate spec reference for !:ValueTuple{} ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.DocAsCode.Metadata.ManagedReference.CSharpNameVisitor.VisitNamedType(INamedTypeSymbol symbol)
   at Microsoft.DocAsCode.Metadata.ManagedReference.NameVisitorCreator.GetName(ISymbol symbol)
   at Microsoft.DocAsCode.Metadata.ManagedReference.CSReferenceItemVisitor.AddLinkItems(INamedTypeSymbol symbol, Boolean withGenericeParameter)
   at Microsoft.DocAsCode.Metadata.ManagedReference.ReferenceItemVisitor.VisitNamedType(INamedTypeSymbol symbol)
   at Microsoft.DocAsCode.Metadata.ManagedReference.ReferenceItemVisitor.VisitNamedType(INamedTypeSymbol symbol)
   at Microsoft.DocAsCode.Metadata.ManagedReference.CompositeYamlModelGenerator.GenerateReferenceInternal(ISymbol symbol, ReferenceItem reference, SymbolVisitorAdapter adapter, Boolean asOverload)
   at Microsoft.DocAsCode.Metadata.ManagedReference.YamlModelGenerator.AddSpecReference(ISymbol symbol, IReadOnlyList`1 typeGenericParameters, IReadOnlyList`1 methodGenericParameters, Dictionary`2 references, SymbolVisitorAdapter adapter)
   at Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.AddSpecReference(ISymbol symbol, IReadOnlyList`1 typeGenericParameters, IReadOnlyList`1 methodGenericParameters)
   --- End of inner exception stack trace ---
   at Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.AddSpecReference(ISymbol symbol, IReadOnlyList`1 typeGenericParameters, IReadOnlyList`1 methodGenericParameters)
   at Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.GetConstantValueForArgumentInfo(TypedConstant arg)
   at Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.GetArgumentInfo(TypedConstant arg)
   at Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.GetArguments(AttributeData attr)
   at Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.<GetAttributeInfo>b__41_3(AttributeData attr)
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.GetAttributeInfo(ImmutableArray`1 attributes)
   at Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.VisitNamedType(INamedTypeSymbol symbol)
   at Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.VisitDescendants[T](IEnumerable`1 children, Func`2 getChildren, Func`2 filter)
   at Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.VisitNamespace(INamespaceSymbol symbol)
   at Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.VisitDescendants[T](IEnumerable`1 children, Func`2 getChildren, Func`2 filter)
   at Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.VisitAssembly(IAssemblySymbol symbol)
   at Microsoft.DocAsCode.Metadata.ManagedReference.RoslynMetadataExtractor.Extract(ExtractMetadataOptions options)
   at Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.GetMetadataFromProjectLevelCache(IBuildController controller, IInputParameters key)
   at Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.<SaveAllMembersFromCacheAsync>d__12.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.<ExtractMetadataAsync>d__10.MoveNext()
        16 Warning(s)
        1 Error(s)
```

**Comment by superyyrrzz** [comment]
> Are we supposed to call nuget restore ourselves before running docfx?

Yes, it is.

> Is docfx supposed to work if nuget restore is not done by handling those ErrorType?

Yes, it's not supposed to fail if not restored. This should be a bug in DocFX.

**Comment by herohua** [comment]
@xen2 , I guess the issue has been resolved by #3228 . Could you kindly retry using the latest version of DocFX? Thanks!
